jdaviz/configs/imviz/tests/test_regions.py

```diff
@@ -0,0 +1,137 @@
+from astropy.io import fits
+
+try:
+    import regions  # noqa
+    HAS_REGIONS = True
+except ImportError:
+    HAS_REGIONS = False
+
+try:
+    import photutils  # noqa
+    HAS_PHOTUTILS = True
+except ImportError:
+    HAS_PHOTUTILS = False
+
+
+class TestLoadStaticRegions:
+    """Test to see if region is loaded.
+    Does not check if region is actually at the correct place in display.
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup_class(self, imviz_app):
+        hdu = fits.ImageHDU(np.zeros((10, 10)), name='SCI')
+
+        # Apply some celestial WCS from
+        # https://learn.astropy.org/rst-tutorials/celestial_coords1.html
+        hdu.header.update({'CTYPE1': 'RA---TAN',
+                           'CUNIT1': 'deg',
+                           'CDELT1': -0.0002777777778,
+                           'CRPIX1': 1,
+                           'CRVAL1': 337.5202808,
+                           'NAXIS1': 10,
+                           'CTYPE2': 'DEC--TAN',
+                           'CUNIT2': 'deg',
+                           'CDELT2': 0.0002777777778,
+                           'CRPIX2': 1,
+                           'CRVAL2': -20.833333059999998,
+                           'NAXIS2': 10})
+
+        # Data with WCS
+        imviz_app.load_data(hdu, data_label='has_wcs')
+
+        # Data without WCS
+        imviz_app.load_data(hdu, data_label='no_wcs')
+        imviz_app.app.data_collection[1].coords = None
+
+        self.imviz = imviz_app
+        self.viewer = imviz_app.app.get_viewer('viewer-1')
+        self.sky = SkyCoord(ra=337.5202808, dec=-20.833333059999998, unit='deg')
+
+    def teardown_method(self, method):
+        # Make sure each test method did not affect interactive region getter
+        # until we implement roundtripping.
+        assert self.imviz.get_interactive_regions() == {}
+
+    def verify_region_loaded(self, region_label, count=2):
+        n = 0
+        for layer in self.viewer.state.layers:
+            if layer.layer.label == region_label:
+                n += 1
+                assert layer.visible
+        assert n == count
+
+    def test_regions_invalid(self):
+        with pytest.raises(TypeError, match='Unsupported region type'):
+            self.imviz.load_static_regions({'reg': self.imviz}, 'has_wcs[SCI,1]')
+
+        # Does not matter if region is invalid here, it is skipped.
+        with pytest.warns(UserWarning, match='is not allowed, skipping'):
+            self.imviz.load_static_regions({'Subset 1': self.imviz}, 'has_wcs[SCI,1]')
+
+        self.verify_region_loaded('reg', count=0)
+        self.verify_region_loaded('Subset 1', count=0)
+
+    @pytest.mark.parametrize('data_label', ('has_wcs[SCI,1]', 'no_wcs[SCI,1]'))
+    def test_regions_mask(self, data_label):
+        mask = np.zeros((10, 10), dtype=np.bool_)
+        mask[0, 0] = True
+        self.imviz.load_static_regions({'my_mask': mask}, data_label)
+        self.verify_region_loaded('my_mask')
```

Comment:
Note for future self: Since Subset is global, I think it is actually loaded 4 times here (twice for each `parametrize` call). But since the region name is the same, the second `parametrize` call probably overwrote the first one. But that is probably okay for testing because we just want to check nothing crashed.